### ISSUE TITLE
How does this code separate HR from LR on val?

### ISSUE THREAD

**Issue by sogangAplus** [zh320/efficient-image-super-resolution-pytorch#3]
Thanks for sharing your code!
 In the sr_dataset.py, I see you distributed val file with scale factor. But I cannot find any code that detects HR image & LR image using image_SRF_upsacle file.
 And I can see that in the sr_base_dataset.py file > __getitem__ -> there is a code for making lr image, and returns both hr & lr image. Which means in each epoch the code makes LR image during validation.
 The conclusion is, I want to know if your code makes LR image during every validation epoch, or if there is a code for separating LR&HR image so that I should put both LR and HR image inside image_SRF_upscale file.
I'll wait for reply :)

**Comment by zh320** [comment]
The reason why it has upscale factor for validation set is because the benchmark datasets have images with different resolution which may not be evenly divided by the upscale factor. Also different implementations for bicubic interpolation may produce slightly different results. In order to have deterministic results, these validation sets are resized to be compatible with different upscale factor using the same interpolation algorithm. If you do not care about benchmarks, there is no need to care about the separation of LR and HR. The LR are just generated from the HR using whatever (bicubic) interpolation unless you want to perform other types of degradation or something like blind super-resolution.

Best

**Comment by oh-chainsmokers** [comment]
Thank you~~
